homeworks/vadym.myroshnyk_VadimMir/homework_5/homework_5.js

```diff
@@ -0,0 +1,70 @@
+// tesk 1
+const someArr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+function sumArrayFunction() {
+    let sum = 0;
+    return someArr.map((item) => {
+        sum = item + sum;
+        return sum;
+    });
+}
+
+console.log(sumArrayFunction());
+
+// tesk 2
+const classes = ['header', 'menu', 'menu_item', 'tabs',
+    'tab_item', 'menu', 'link', 'tabs', 'tab_item', 'menu',
+    'menu_item', 'menu', 'menu_item'];
+
+function returnObjKey() {
+    const result = {};
+
+    classes.reduce((previousValue, value, i) => {
+        let sumValue = 0;
+        sumValue = classes.filter(item => item === classes[i]).length;
+        result[value] = sumValue;
+        return result[value];
```

Comment:
looping over same array twice when not needed, wrong

Reply:
seems like you do not understand what are you doing here...

you trigger `classes.reduce()` then you trigger `classes.filter()` in each iteration of `reduce()`, read about what reduce is doing, how it's accumulator is being passed into each iteration... this will help you